### ISSUE TITLE
Ability to share a post or comment using a specific front end.

### ISSUE THREAD

**Issue by K4LCIFER** [thunder-app/thunder#1424]
### Is Your Feature Request Related to a Problem?

Sharing a post or comment appears to use the Lemmy server's base URL, so it uses the front end that is attached to that base URL (typically Lemmy-UI); however, some Lemmy servers offer alternative UI's, e.g. Photon, Alexandrite, etc. 

### Feature Description

It would be nice to be able to share a post using a specific front end that the server offers.

### Additional Context

This may not be possible — there is likely no API method to determine this. Afaik, this information is usually added in the sidebar of a server.

**Comment by hjiangsu** [comment]
> This may not be possible — there is likely no API method to determine this. Afaik, this information is usually added in the sidebar of a server.

Yup, as you mentioned here, this likely won't be possible for us to do unless it comes from the API, or we 
- manually add entries for each and every instance that has an alternative front-end
- parse the instance sidebar and detect alternative front-ends

**Comment by K4LCIFER** [comment]
Would it perhaps be possible for the user to specify it themselves? There could be an option in the settings for a user to specify the link to the front end that they wish to share with. For example, if someone is on lemmy.world, and they want to use the Photon UI at photon.lemmy.world, they could simply specify in the settings that they wish to share through that link rather than the lemmy.world link.  The API requests would be the exact same, afaik, just the site that responds would be the specified front end.

This would also have the double effect of allowing a user to specify which instance that they wish to default to when sharing posts — there could be a scenario where a user doesn't ever want to share posts from their home server, but, rather, some other proxy server.
